blscurve/eth2_keygen/blst_vect.nim

```diff
@@ -0,0 +1,57 @@
+## BEWARE!
+##
+## This module exists as a work-around for the following issue in the BLST library:
+## https://github.com/supranational/blst/issues/29
+##
+## Since it's not possible to import vect.h and blst.h in the same compilation unit,
+## we have to be careful to avoid producing imports to vect.h anywhere outside of
+## this module.
+##
+## In other words, this module essentially wraps all functions that we want to use
+## from the vect.h header in Nim functions that the Nim compiler knows how to link
+## accross module boundaries, thus eliminating the need for any other compilation
+## unit to depend on the vect.h header.
+
+import
+  os
+
+type
+  limb_t = uint64
+  vec256 = array[4, limb_t]
+  vec512 = array[8, limb_t]
+
+const srcPath = currentSourcePath.parentDir & "/../../vendor/blst/src"
+
+# XXX This was copied from hkdf_mod_r_blst.nim ithout much analysis
```

Comment:
did you check the performance benchmarks? ie these were important at some point cc @mratsim

Reply:
This won't impact perf.

Also with BLST offering builtin EIP2333 support we can actually remove the hkdf mod r proc: https://github.com/supranational/blst/commit/4e1935eb722289789f5b8f8447415f77b63ff37c

(no official release that include that commit though)